<div style="border:solid green 2px; padding: 20px"> <h1 style="color:green; margin-bottom:20px">Комментарий наставника</h1>

Привет! Далее в файле мои коммефнтарии ты сможешь найти в ячейках, аналогичных данной ( если рамки комментария зелёные - всё сделано правильно; жёлтые - есть замечания, но не критично; красные - нужно переделать). Не удаляй эти комментарии и постарайся учесть их в ходе выполнения проекта. 

<h1>Сборный проект - 2 </h1>

<h2>Описание проекта</h2>
<p>Вы работаете в стартапе, который продаёт продукты питания. Нужно разобраться, как ведут себя пользователи вашего мобильного приложения.</p>
<p>Изучите воронку продаж. Узнайте, как пользователи доходят до покупки. Сколько пользователей доходит до покупки, а сколько — «застревает» на предыдущих шагах? На каких именно?</p>
<p>После этого исследуйте результаты A/B-эксперимента. Дизайнеры захотели поменять шрифты во всём приложении, а менеджеры испугались, что пользователям будет непривычно. Договорились принять решение по результатам A/B-теста. Для него пользователей разбили на 3 группы: 2 контрольные со старыми шрифтами и одну экспериментальную — с новыми. Выясните, какой шрифт лучше.</p>
<p>В случае общей аналитики и A/B-эксперимента работайте с одними и теми же данными. В реальных проектах всегда идут эксперименты. Аналитики исследуют качество работы приложения по общим данным, не учитывая принадлежность пользователей к экспериментам.</p>

<h2>Описание по выполнению проекта</h2>

<b>Шаг 1. Загрузите данные и подготовьте их к анализу</b>

<b>Шаг 2. Подготовьте данные</b>
<li>Замените названия столбцов на удобные для вас;</li>
<li>Проверьте пропуски и типы данных. Откорректируйте, если нужно;</li>
<li>Добавьте столбец даты и времени, а также отдельный столбец дат;</li>

<b>Шаг 3. Изучите и проверьте данные</b>
<li>Сколько всего событий в логе?</li>
<li>Сколько всего пользователей в логе?</li>
<li>Сколько в среднем событий приходится на пользователя?</li>
<li>Данными за какой период вы располагаете? Найдите максимальную и минимальную дату. Постройте гистограмму по дате и времени. Можно ли быть уверенным, что у вас одинаково полные данные за весь период? Технически в логи новых дней по некоторым пользователям могут «доезжать» события из прошлого — это может «перекашивать данные». Определите, с какого момента данные полные и отбросьте более старые. Данными за какой период времени вы располагаете на самом деле?</li>
<li>Много ли событий и пользователей вы потеряли, отбросив старые данные?</li>
<li>Проверьте, что у вас есть пользователи из всех трёх экспериментальных групп.</li>

<b>Шаг 4. Изучите воронку событий</b>
<li>Посмотрите, какие события есть в логах, как часто они встречаются. Отсортируйте события по частоте.</li>
<li>Посчитайте, сколько пользователей совершали каждое из этих событий. Отсортируйте события по числу пользователей. Посчитайте долю пользователей, которые хоть раз совершали событие.</li>
<li>Предположите, в каком порядке происходят события. Все ли они выстраиваются в последовательную цепочку? Их не нужно учитывать при расчёте воронки.</li>
<li>По воронке событий посчитайте, какая доля пользователей проходит на следующий шаг воронки (от числа пользователей на предыдущем). То есть для последовательности событий A → B → C, посчитайте отношение числа пользователей с событием B к количеству пользователей с событием A.</li>
<li>На каком шаге теряете больше всего пользователей?</li>
<li>Какая доля пользователей доходит от первого события до оплаты?</li>

<b>Шаг 5. Изучите результаты эксперимента</b>
<li>Сколько пользователей в каждой экспериментальной группе?</li>
<li>Есть 2 контрольные группы для А/А-эксперимента, чтобы проверить корректность всех механизмов и расчётов. Проверьте, находят ли статистические критерии разницу между выборками 246 и 247.</li>
<li>Выберите самое популярное событие. Посчитайте число пользователей, совершивших это событие в каждой из контрольных групп. Посчитайте долю пользователей, совершивших это событие. Проверьте, будет ли отличие между группами статистически достоверным. Проделайте то же самое для всех других событий (удобно обернуть проверку в отдельную функцию). Можно ли сказать, что разбиение на группы работает корректно?</li>
<li>Аналогично поступите с группой с изменённым шрифтом. Сравните результаты с каждой из контрольных групп в отдельности по каждому событию. Сравните результаты с объединённой контрольной группой. Какие выводы из эксперимента можно сделать?</li>
<li>Какой уровень значимости вы выбрали при проверке статистических гипотез выше? Посчитайте, сколько проверок статистических гипотез вы сделали. При уровне значимости 0.1 каждый десятый раз можно получать ложный результат. Какой уровень значимости стоит применить? Если вы хотите изменить его, проделайте предыдущие пункты и проверьте свои выводы.</li>

<big><big>Описание данных</big></big>

Путь к файлу: <b>/datasets/logs_exp.csv</b>:
        
Каждая запись в логе — это действие пользователя, или событие.
<li><b>EventName</b> — название события;</li>
<li><b>DeviceIDHash</b> **— уникальный идентификатор пользователя;</li>
<li><b>EventTimestamp</b> — время события;</li>
<li><b>ExpId</b> — номер эксперимента: 246 и 247 — контрольные группы, а 248 — экспериментальная.
</li>

<h1>Шаг 1. Загрузка данных и подготовьте их к анализу</h1>

In [32]:
import pandas as pd 
import numpy as np
from numpy import median
import math as mth
import seaborn as sns
import datetime as dt
%matplotlib inline
import matplotlib.pyplot as plt
from scipy import stats as st
import plotly.express as px
from plotly import graph_objects as go
from plotly.offline import iplot

In [33]:
logs = pd.read_csv('/datasets/logs_exp.csv', delimiter='\t')

In [34]:
logs.head(10)

EventName         DeviceIDHash  EventTimestamp  ExpId
0         MainScreenAppear  4575588528974610257      1564029816    246
1         MainScreenAppear  7416695313311560658      1564053102    246
2  PaymentScreenSuccessful  3518123091307005509      1564054127    248
3         CartScreenAppear  3518123091307005509      1564054127    248
4  PaymentScreenSuccessful  6217807653094995999      1564055322    248
5         CartScreenAppear  6217807653094995999      1564055323    248
6       OffersScreenAppear  8351860793733343758      1564066242    246
7         MainScreenAppear  5682100281902512875      1564085677    246
8         MainScreenAppear  1850981295691852772      1564086702    247
9         MainScreenAppear  5407636962369102641      1564112112    246

<h1>Шаг 2. Подготовьте данные</h1>

<h3>Замените названия столбцов</h3>

In [35]:
logs.columns = ['event_name', 'user_id', 'event_time', 'test_group']
logs.head(5)

event_name              user_id  event_time  test_group
0         MainScreenAppear  4575588528974610257  1564029816         246
1         MainScreenAppear  7416695313311560658  1564053102         246
2  PaymentScreenSuccessful  3518123091307005509  1564054127         248
3         CartScreenAppear  3518123091307005509  1564054127         248
4  PaymentScreenSuccessful  6217807653094995999  1564055322         248

In [36]:
logs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244126 entries, 0 to 244125
Data columns (total 4 columns):
event_name    244126 non-null object
user_id       244126 non-null int64
event_time    244126 non-null int64
test_group    244126 non-null int64
dtypes: int64(3), object(1)
memory usage: 7.5+ MB


<li>Столбец и event_time имеет не удобный для нас тип данных. Заменим его на тип datatime, а так же в отдельный столбец вынесем информацию о просто дате.</li>

In [37]:
logs.event_time = pd.to_datetime(logs.event_time, unit = 's')
logs['data'] = logs.event_time.dt.date
logs['data'] = logs['event_time'].dt.date
logs['data'] = pd.to_datetime(logs['data'], format='%Y-%m-%d')

In [38]:
logs.isnull().sum()

event_name    0
user_id       0
event_time    0
test_group    0
data          0
dtype: int64

In [39]:
logs.duplicated().sum()

413

In [40]:
logs.drop_duplicates().reset_index(drop = True)

event_name              user_id          event_time  \
0              MainScreenAppear  4575588528974610257 2019-07-25 04:43:36   
1              MainScreenAppear  7416695313311560658 2019-07-25 11:11:42   
2       PaymentScreenSuccessful  3518123091307005509 2019-07-25 11:28:47   
3              CartScreenAppear  3518123091307005509 2019-07-25 11:28:47   
4       PaymentScreenSuccessful  6217807653094995999 2019-07-25 11:48:42   
...                         ...                  ...                 ...   
243708         MainScreenAppear  4599628364049201812 2019-08-07 21:12:25   
243709         MainScreenAppear  5849806612437486590 2019-08-07 21:13:59   
243710         MainScreenAppear  5746969938801999050 2019-08-07 21:14:43   
243711         MainScreenAppear  5746969938801999050 2019-08-07 21:14:58   
243712       OffersScreenAppear  5746969938801999050 2019-08-07 21:15:17   

        test_group       data  
0              246 2019-07-25  
1              246 2019-07-25  
2              248 2019-07-25  
3              248 2019-07-25  
4              248 2019-07-25  
...            ...        ...  
243708         247 2019-08-07  
243709         246 2019-08-07  
243710         246 2019-08-07  
243711         246 2019-08-07  
243712         246 2019-08-07  

[243713 rows x 5 columns]

In [41]:
logs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244126 entries, 0 to 244125
Data columns (total 5 columns):
event_name    244126 non-null object
user_id       244126 non-null int64
event_time    244126 non-null datetime64[ns]
test_group    244126 non-null int64
data          244126 non-null datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(1)
memory usage: 9.3+ MB


<li>Теперь столбец и event_time имеет тип данных datetime64[ns], так же как и столбец data. Пропущенных значений в данных нет, зато есть дубликаты. Они были удалены, так как их всего 413, а строк в датасете 244126. Не ощутимые потери.</li>

In [42]:
events = len(logs)
users = logs['user_id'].nunique()
events_per_users = events/users
data_start = logs['data'].min()
data_finish = logs['data'].max()
print('Всего событий в логе: {}'.format(events))
print('Всего прользователей в логе: {}'.format(users))
print('Среднее кол-во событий на пользователя: {}'.format(events_per_users))
print('Старт отслеживания воронки продаж в данных: {}'.format(data_start))
print('Конец отслеживания воронки продаж в данных: {}'.format(data_finish))

Всего событий в логе: 244126
Всего прользователей в логе: 7551
Среднее кол-во событий на пользователя: 32.33028737915508
Старт отслеживания воронки продаж в данных: 2019-07-25 00:00:00
Конец отслеживания воронки продаж в данных: 2019-08-07 00:00:00


In [43]:
data_time_hist = logs.groupby('data')['event_time'].count().reset_index()
fig = px.bar(data_time_hist, x="data", y="event_time", title='Распределение количества событий по дням')
fig.show()

<li>На гистрограмме видно, что наиболее полные данные у нас начинаются с 1 августа. Данными с 1 по 7 августа актуальны. С ними и продолжим работу. Данные, которые раньше 1го числа дальше анализироваться не будут.</li>

In [44]:
logs_new = logs.query('data >= "2019-08-01"')
logs_new['data'].unique()

array(['2019-08-01T00:00:00.000000000', '2019-08-02T00:00:00.000000000',
       '2019-08-03T00:00:00.000000000', '2019-08-04T00:00:00.000000000',
       '2019-08-05T00:00:00.000000000', '2019-08-06T00:00:00.000000000',
       '2019-08-07T00:00:00.000000000'], dtype='datetime64[ns]')

In [45]:
events_new = len(logs_new)
users_new = logs_new['user_id'].nunique()
differents_events = events - events_new
differents_users = users - users_new
print('Кол-во событий в логе до фильтрации: {}'.format(events))
print('Всего прользователей в логе до фильтрации: {}'.format(users))
print('Кол-во событий в логе после фильтрации: {}'.format(events_new))
print('Всего прользователей в логе после фильтрации: {}'.format(users_new))
print('Разница в кол-ве событий: {}'.format(differents_events))
print('Разница в кол-ве пользователей: {}'.format(differents_users))

Кол-во событий в логе до фильтрации: 244126
Всего прользователей в логе до фильтрации: 7551
Кол-во событий в логе после фильтрации: 241298
Всего прользователей в логе после фильтрации: 7534
Разница в кол-ве событий: 2828
Разница в кол-ве пользователей: 17


<div style="border:solid green 2px; padding: 20px"> <h1 style="color:green; margin-bottom:20px">Комментарий наставника</h1>

Хорошо, все верно рассчитано. Только лучше считать процент удаленных, а не абсолютные цифры. 

In [46]:
user_id = logs_new.groupby(['test_group', 'event_name'])['event_time'].count().reset_index()
user_id

test_group               event_name  event_time
0          246         CartScreenAppear       14711
1          246         MainScreenAppear       37708
2          246       OffersScreenAppear       14773
3          246  PaymentScreenSuccessful       11910
4          246                 Tutorial         323
5          247         CartScreenAppear       12456
6          247         MainScreenAppear       39123
7          247       OffersScreenAppear       15182
8          247  PaymentScreenSuccessful       10043
9          247                 Tutorial         343
10         248         CartScreenAppear       15198
11         248         MainScreenAppear       40600
12         248       OffersScreenAppear       16395
13         248  PaymentScreenSuccessful       12160
14         248                 Tutorial         373

In [47]:
fig = go.Figure(data = [go.Pie(labels=user_id['test_group'], values = user_id['event_time'])])
fig.show()

<li>Взяли данные с 1 по 7 августа. В каждой группе у нас есть люди, в каждой из воронок они так же сохранились.</li>

In [48]:
event_name = logs_new.groupby('event_name')['event_time'].count().reset_index()
(event_name.sort_values('event_time',  ascending=False).style
                       .bar(subset=['event_time'], color='#B0E0E6', vmin=0, width=100)
                       .set_caption("Распределение событий в логах")
)

<li>Реже всех встречается событие "Tutorial", то есть обучение, скорее всего его пропускают, при посещении сайта. Так что для дальнейшего анализа его можно пропустить, так как не влияет, дойдет ли до финальной покупки.</li>

In [49]:
user_id_count = logs_new.groupby(['event_name'])['user_id'].nunique().reset_index()
events_users_count = event_name.merge(user_id_count, on='event_name', how='inner')
events_users_count.columns = ['event_name', 'event_time', 'users_count']
events_users_count['percent'] = events_users_count['users_count'].apply(lambda x: round(x/7419*100, 2))
(events_users_count.sort_values('event_time',  ascending=False).style
                       .bar(subset=['users_count','percent'], color='#B0E0E6', vmin=0, width=100)
                       .set_caption("События по числу пользователей и процентное соотношение")
)

<div style="border:solid yellow 2px; padding: 20px"> <h1 style="color: yellow; margin-bottom:20px">Комментарий наставника</h1>

Хорошо, только стоило делить на общее количество пользователей, а их побольше, чем пользователей на первом шаге (7517). 

<li>Как писалось выше, событие "Tutorial" не является обязательным, с ним дальше можно не работать. Так что исключим его из данных</li>

In [50]:
logs_finel = logs_new[logs_new['event_name'] != 'Tutorial']
logs_finel.head()

event_name              user_id          event_time  test_group  \
2829    MainScreenAppear  3737462046622621720 2019-08-01 00:08:00         246   
2830    MainScreenAppear  3737462046622621720 2019-08-01 00:08:55         246   
2831  OffersScreenAppear  3737462046622621720 2019-08-01 00:08:58         246   
2832    MainScreenAppear  1433840883824088890 2019-08-01 00:08:59         247   
2833    MainScreenAppear  4899590676214355127 2019-08-01 00:10:15         247   

           data  
2829 2019-08-01  
2830 2019-08-01  
2831 2019-08-01  
2832 2019-08-01  
2833 2019-08-01

<li>MainScreenAppear - появляется главный экран</li>
<li>Tutorial - прохождение обучения по использованию (по желанию, так как экран не обязательный, мы его исключили из анализа)</li>
<li>OffersScreenAppear - переход на экран предложений</li>
<li>CartScreenAppear - появляется экран корзины</li>
<li>PaymentScreenSuccessful - экран успешной оплаты</li>

In [51]:
fig = go.Figure(go.Funnel(
    y = ["Main Screen Appear", "Offers Screen Appear", "Cart Screen Appear", "Payment Screen Successful"],
    x = [7419, 4593, 3734, 3539]))

fig.show()

<li>Всего 18,4% пользователей от общего числа доходят до события оплаты. Большое всего теряется пользователей на переходе с первого на второй этап. Из 100% туда попадают только 62%.</li>

<div style="border:solid green 2px; padding: 20px"> <h1 style="color:green; margin-bottom:20px">Комментарий наставника</h1>

Хорошо. 

In [52]:
test_group = logs_finel.groupby('test_group')['user_id'].nunique().reset_index()
test_group

test_group  user_id
0         246     2483
1         247     2512
2         248     2535

<li>Номер эксперимента: 246 и 247 — контрольные группы, а 248 — экспериментальная.</li>

In [53]:
test_group_by_events = logs_finel.pivot_table(index = 'test_group',
                                   columns = 'event_name',
                                   values = 'user_id',
                                   aggfunc = 'nunique')#.reset_index()
test_group_by_events

event_name  CartScreenAppear  MainScreenAppear  OffersScreenAppear  \
test_group                                                           
246                     1266              2450                1542   
247                     1238              2476                1520   
248                     1230              2493                1531   

event_name  PaymentScreenSuccessful  
test_group                           
246                            1200  
247                            1158  
248                            1181

In [54]:
test_group_by_events = test_group_by_events.merge(test_group, on='test_group', how='inner')
test_group_by_events = test_group_by_events.rename(columns={'user_id': 'users_count'})
test_group_by_events = test_group_by_events.set_index('test_group')
test_group_by_events

CartScreenAppear  MainScreenAppear  OffersScreenAppear  \
test_group                                                           
246                     1266              2450                1542   
247                     1238              2476                1520   
248                     1230              2493                1531   

            PaymentScreenSuccessful  users_count  
test_group                                        
246                            1200         2483  
247                            1158         2512  
248                            1181         2535

In [55]:
test_group_by_events[test_group_by_events.index.duplicated()]

Empty DataFrame
Columns: [CartScreenAppear, MainScreenAppear, OffersScreenAppear, PaymentScreenSuccessful, users_count]
Index: []

In [56]:
new_row = test_group_by_events.loc[246] + test_group_by_events.loc[247]
new_row.name = '246+247'
test_group_by_events = test_group_by_events.append([new_row])
test_group_by_events = test_group_by_events[['MainScreenAppear', 'OffersScreenAppear',
                                   'CartScreenAppear', 'PaymentScreenSuccessful', 'users_count']]
test_group_by_events = test_group_by_events.reindex([246, 247, '246+247', 248])
test_group_by_events

MainScreenAppear  OffersScreenAppear  CartScreenAppear  \
246                  2450                1542              1266   
247                  2476                1520              1238   
246+247              4926                3062              2504   
248                  2493                1531              1230   

         PaymentScreenSuccessful  users_count  
246                         1200         2483  
247                         1158         2512  
246+247                     2358         4995  
248                         1181         2535

<li>Из имебщихся данных собрали таблицу test_group_by_events, в котрой информация по каждой из тестовых групп, а так же информация по сумме участников 246+247 групп. В колонках 'MainScreenAppear', 'OffersScreenAppear', 'CartScreenAppear', 'PaymentScreenSuccessful' инфорация сколько всего участников на каждом этапе из каждой группы, "users_count" - сколько всего участников в группе.</li>

Гипотеза:

H0: между долями нет значимой разница
H1: между долями есть значимая разница
Z-тест (z-критерий Фишера) — класс методов статистической проверки гипотез (статистических критериев), основанных на нормальном распределении. Обычно применяется для проверки равенства средних значений при известной дисперсии генеральной совокупности или при оценке выборочного среднего стандартизованных значений. С помощью него проверим гипотезы.

Возьмем уровень значимости P-value = 0,005, чтобы снизить вероятность ошибки

In [58]:
#напишем функцию для проверки гипотез
def check_hypothesis(part0, part1, total0, total1):  
    alpha = .05 
    # пропорция успехов в первой группе:
    p1 = part0 / total0 
    # пропорция успехов во второй группе:
    p2 = part1 / total1
    # пропорция успехов в комбинированном датасете:
    p_combined = (part0 + part1) / (total0 + total1)
    # разница пропорций в датасетах
    difference = p1 - p2
    # считаем статистику в ст.отклонениях стандартного нормального распределения
    z_value = difference / mth.sqrt(p_combined * (1 - p_combined) * (1/total0 + 1/total1))
    # задаем стандартное нормальное распределение (среднее 0, ст.отклонение 1)
    distr = st.norm(0, 1) 
    p_value = (1 - distr.cdf(abs(z_value))) * 2
    return p_value

<div style="border:solid green 2px; padding: 20px"> <h1 style="color:green; margin-bottom:20px">Комментарий наставника</h1>

Функция для проверки написана верно. 

In [59]:
AA_pivot = pd.DataFrame(columns=['246_А', '247_А', 'Событие', 'Принимаемая гипотеза'])

result_options = ["Отвергаем нулевую гипотезу - доли разные",
                  "Между долями нет значимой разницы"]

group_1 = 246
current_row = test_group_by_events.loc[group_1]
group_2 = 247
for event_number in range(4):
    curr_event = test_group_by_events.columns[event_number]
    p_value_result = check_hypothesis(current_row[curr_event], 
                                   test_group_by_events[curr_event][group_2],
                                   current_row['users_count'],
                                   test_group_by_events['users_count'][group_2]
                           )
    new_row = {'246_А': group_1,
               '247_А': group_2,
               'Событие': test_group_by_events.columns[event_number],
               'Принимаемая гипотеза': result_options[p_value_result >= 0.05]}
    AA_pivot = AA_pivot.append([new_row])

AA_pivot.index = range(len(AA_pivot))
AA_pivot

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:19: DeprecationWarning:

In future, it will be an error for 'np.bool_' scalars to be interpreted as an index



246_А 247_А                  Событие               Принимаемая гипотеза
0   246   247         MainScreenAppear  Между долями нет значимой разницы
1   246   247       OffersScreenAppear  Между долями нет значимой разницы
2   246   247         CartScreenAppear  Между долями нет значимой разницы
3   246   247  PaymentScreenSuccessful  Между долями нет значимой разницы

<li>Это значит, что статистически значимых различий между группами не обнаружено, информация корректна. Можно дальше продолжать исследоваение</li>

Гипотеза:

H0: между контрольной группой и экспереминтальной нет статистически значимой разницы
H1: между контрольной группой и экспереминтальной есть значимая разница

In [61]:
AB_pivot = pd.DataFrame(columns=['Контрольные группы', 'Экспериментальная группа', 'Событие', 'Принимаемая гипотеза'])

result_options = ["Отвергаем нулевую гипотезу - доли разные",
                  "Нет статистически значимой разницы"]

for i in range(3):
    group_1 = test_group_by_events.index[i]
    current_row = test_group_by_events.loc[group_1]
    group_2 = 248
    for event_number in range(4):
        curr_event = test_group_by_events.columns[event_number]
        p_value_result = check_hypothesis(current_row[curr_event], 
                                       test_group_by_events[curr_event][group_2],
                                       current_row['users_count'],
                                       test_group_by_events['users_count'][group_2]
                               )
        new_row = {'Контрольные группы': group_1,
                   'Экспериментальная группа': group_2,
                   'Событие': test_group_by_events.columns[event_number],
                   'Принимаемая гипотеза': result_options[p_value_result >= 0.05]}
        AB_pivot = AB_pivot.append([new_row])

AB_pivot.index = range(len(AB_pivot))
AB_pivot

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:20: DeprecationWarning:

In future, it will be an error for 'np.bool_' scalars to be interpreted as an index



Контрольные группы Экспериментальная группа                  Событие  \
0                 246                      248         MainScreenAppear   
1                 246                      248       OffersScreenAppear   
2                 246                      248         CartScreenAppear   
3                 246                      248  PaymentScreenSuccessful   
4                 247                      248         MainScreenAppear   
5                 247                      248       OffersScreenAppear   
6                 247                      248         CartScreenAppear   
7                 247                      248  PaymentScreenSuccessful   
8             246+247                      248         MainScreenAppear   
9             246+247                      248       OffersScreenAppear   
10            246+247                      248         CartScreenAppear   
11            246+247                      248  PaymentScreenSuccessful   

                  Принимаемая гипотеза  
0   Нет статистически значимой разницы  
1   Нет статистически значимой разницы  
2   Нет статистически значимой разницы  
3   Нет статистически значимой разницы  
4   Нет статистически значимой разницы  
5   Нет статистически значимой разницы  
6   Нет статистически значимой разницы  
7   Нет статистически значимой разницы  
8   Нет статистически значимой разницы  
9   Нет статистически значимой разницы  
10  Нет статистически значимой разницы  
11  Нет статистически значимой разницы

<li>Статистически значимых различий ни в одном из событий не обнаружено.</li>

<div style="border:solid green 2px; padding: 20px"> <h1 style="color:green; margin-bottom:20px">Комментарий наставника</h1>

Проверка гипотез выполнена верно. 

Удалось выяснить, что у нас есть данные с 25 июля по 7 августа. Но при этом наиболее полные данные только с 1 августа. Так что в основу анализа легли данные с 1 по 7 августа.

Порядок событий удалось установить такой:

MainScreenAppear
Tutorial
OffersScreenAppear
CartScreenAppear
PaymentScreenSuccessful
Однако, этап "Tutorial" очень мало кто посещал, и он не является обязательным, а так же никак не влияет на нашу поставленную задачу, так что дальнейший анализ продолжался без этого этапа.

Этап, на котором больше всего теряется пользователей,это переход со страницы MainScreenAppear к странице OffersScreenAppear. Из 100% туда попадают только 62%. Всего 18,4% пользователей от общего числа доходят до события оплаты.

Что касается контрольных и экспериментальной групп, то появление нового шрифта никак статистически не подтвердило улучшение или ухудшение продаж. Вообще никак не повлияло на поведение пользователей. Но этот вывод удалось сделать из тех данных, которые нам предоставлены. Одна неделя - весьма не большой срок для выводов. Стоит рекомендовать команде продлить эксперимент еще на 1 или 2 недели с тем же числом пользователей.

<div style="border:solid green 2px; padding: 20px"> <h1 style="color:green; margin-bottom:20px">Комментарий наставника</h1>

Проект выполнен отлично, прекрассное завершение модуля, молодец. 